---
layout: post
title: Learning Timeline for Clinical Bioinformatics
excerpt:
categories: ClinicalBioinformatics
tags: ["Clinical Bioinformatics", "Introduction", "Resources", "Learning", "Time", "Timeline"]
published: true
comments: true
share: true
---

## How much time does learning clinical bioinformatics take?

I thought it might be helpful to provide a general idea of how much time would be required to acquire specific clinical bioinformatics skills. The following time estimates are quite rough and the time requirement will vary greatly between different individuals.

Learning is a continuous process, but if you were to add up together all the bits and pieces of time dedicated to a specific process, I think the following estimates might roughly capture the time requirements:

* Git/Github: 1-2 days
* Installing tools from the terminal: 2-3 days
* Install R language + RStudio + install some useful libraries: 0.5-2 days  
* Install Python language + pick an editor + install some useful: 1-2 days

### Don't forget the statistics

Most genomics researchers seem to wish they had a better grasp on statistics. A solid foundation in statistics is required for hypothesis formulation and testing when handling big data.

Some excellent online courses are available on [Coursera](https://www.coursera.org/), such as [Statistics with R Specialization](https://www.coursera.org/specializations/statistics). Open online course platforms have been gaining popularity in the recent years and many prestigious universities, including John Hopkins University and Stanford University, offer online courses on such platforms.

### Learning coding

First of all, it is most helpful to clarify what are your needs and why you would like to use coding for ! Although most people could learn the basics of R or Python in 2-3 weeks, this wouldn't necessarily be useful on a day-to-day basis for any given researcher. Learning coding is a continuous process and lifelong learning journey, and understanding the theory behind algorithm might be more useful than acquiring technical skills for you.

### User interface: so many excellent and useful tools are out there

There are so many useful (and often free) tools available on the web. If you are a clinical trainee or a clinician, becoming familiar with these tools might well be the best investment of your time. I would recommend you think about your typical week and write down what are the repetitive tasks you are performing. While you have identified these, try to clear at least 1-2h of your schedule to Google your question and add terms such as "web resources", "web-based", "online tools", "software" or "automated".

**TIP** You might find it useful to become familiar with txt file format instead of word documents. For example, if you are storing a list of genes associated with a particular phenotype, I recommend that you use either a plain txt or csv format because it will be easier to copy-paste a list which is already in the "right format" into web applications "search bar" and will be faster than entering manually a list of genes.

**Phenotype**

For example, say you find yourself entering the same combination of phenotypic terms over and over again in different genomic web resources, google "phenotype+software" and you fill find the [Phenotips](https://phenotips.org/) (Girdea et al, 2013)[1]. Phenotips is not only a web application, but also a software you can download and install locally on your computer. If approved by the local health authorities/hospital privacy policies, one can use PhenoTips to enter patient identifying data and search for Mendelian disorders matching the described phenotype.

I would recommend to get into the habit of learning about one new tool per week. There are so many tools nowadays

***

#### References

[1] Girdea, M., Dumitriu, S., Fiume, M., Bowdin, S., Boycott, K. M., Chénier, S., Chitayat, D., Faghfoury, H., Meyn, M. S., Ray, P. N., So, J., Stavropoulos, D. J. and Brudno, M. (2013), PhenoTips: Patient Phenotyping Software for Clinical and Research Use. Hum. Mutat., 34: 1057–1065. doi: 10.1002/humu.22347
